force-app/classes/test/MethodSpyTest.cls

```diff
@@ -80,6 +80,25 @@ private class MethodSpyTest {
     }
   }
 
+  @IsTest
+  static void givenSpyConfiguredToThenThrowException_whenCallingTheSpyWithMatchingParams_itThrowsTheException() {
+    // Arrange
+    final MethodSpy sut = new MethodSpy('methodName');
+    final Exception expectedException = new IllegalArgumentException(
+      'test exception'
+    );
+    sut.whenCalledWith('Expected Param').thenThrow(expectedException);
+
+    try {
+      // Act
+      sut.call(new List<Object>{ 'Expected Param' });
+      System.assert(false, 'test should have throw an exception');
+    } catch (Exception e) {
+      // Assert
+      System.assertEquals(expectedException.getMessage(), e.getMessage());
```

Comment:
**suggestion(_nit_):** we coul add the message third parameter of `System.assertEquals` call